source/CIGameScene.cpp

```diff
@@ -264,3 +268,32 @@ void GameScene::addStardust(const Size bounds) {
     _stardustContainer->addStardust(c, bounds);
     
 }
+
+/**
+ * This method applies the power ups of special stardust.
+ *
+ * @param bounds the bounds of the game screen
+ * @param stardustQueue the stardustQueue
+ */
+void GameScene::processSpecialStardust(const cugl::Size bounds, const std::shared_ptr<StardustQueue> stardustQueue) {
+    std::vector<std::shared_ptr<StardustModel>> powerupQueue = stardustQueue->getPowerupQueue();
+    for (size_t ii = 0; ii < powerupQueue.size(); ii++) {
+        std::shared_ptr<StardustModel> stardust = powerupQueue[ii];
+
+        switch (stardust->getStardustType()) {
+            case StardustModel::Type::METEOR:
+                CULog("METEOR SHOWER!");
+                stardustQueue->addStardust(stardust->getColor(), bounds);
+                stardustQueue->addStardust(stardust->getColor(), bounds);
+                stardustQueue->addStardust(stardust->getColor(), bounds);
+                stardustQueue->addStardust(stardust->getColor(), bounds);
+                stardustQueue->addStardust(CIColor::getRandomColor(), bounds);
+                stardustQueue->addStardust(CIColor::getRandomColor(), bounds);
```

Comment:
maybe we should add like a percent chance model to this if we ever decide to change it

Reply:
Do you mean like percent chance for colors or number of stardust? or both?

Reply:
either or works